server/src/main/java/org/opensearch/search/query/ConcurrentQueryPhaseSearcher.java

```diff
@@ -39,6 +41,68 @@ public class ConcurrentQueryPhaseSearcher extends DefaultQueryPhaseSearcher {
      */
     public ConcurrentQueryPhaseSearcher() {}
 
+    @Override
+    public boolean searchWith(
+        SearchContext searchContext,
+        ContextIndexSearcher searcher,
+        Query query,
+        LinkedList<QueryCollectorContext> collectors,
+        boolean hasFilterCollector,
+        boolean hasTimeout
+    ) throws IOException {
+        // Fast path - skip extension logic entirely if no extensions are registered
+        List<QueryPhaseExtension> extensions = queryPhaseExtensions();
+        if (extensions == null || extensions.isEmpty()) {
```

Comment:
Nitpick, but is `null` a valid value? I'd prefer to just let the code NPE if a plugin overrode this method and returned `null`.

Also, another nitpick, but this feels a bit like a premature optimization. Is checking for an empty list here really that much faster than the two for loops below for an empty list? I'd prefer to have less code and few code paths unless this is clearly faster in practice.